Docs - add sections on configuring page templates & key terms

emcarru opened the pull request.

domtra commented on README.md:
````diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
+Flynt\echoHtmlFromConfigFile('default.json');
+```
+
+The JSON template configuration files are found in `config/templates`. These configuration files define the [areas](#areas) and [components](#components) which are loaded into the template.
````
i'd rather say `files define the components and their areas which are loaded`

domtra commented on README.md:
```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
+
+The JSON template configuration files are found in `config/templates`. These configuration files define the [areas](#areas) and [components](#components) which are loaded into the template.
+
+Take `config/templates/default.json` as an example. This template calls the `DocumentDefault` component and registers one area: `layout`. The `layout` area calls the `LayoutSinglePost` component, which in turn contains three nested [areas](#areas): `mainHeader`, `pageComponents`, and `mainFooter`. In addition, the `pageComponents` area calls the `ComponentLoaderFlexible` component.
```
i am not really happy with `calls` and `registers`. components are registered somewhere else. and not called. What about contains, consists of, is made up of / is composed with?. I am sure you have some ideas ;)

domtra commented on README.md:
````diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
+}
+```
+
+To complete the picture, the `layout` area defined in the template is then output in `Components/DocumentDefault/index.twig`:
````
this sentence confuses me

domtra commented on README.md:
```diff
@@ -43,12 +48,94 @@ In your terminal, navigate to `<your-project>/wp-content/themes/flynt-starter-th
+### Features
+With WordPress, it is easy to create one large `functions.php` file, crammed full of all the custom logic your theme may need. This can get messy. In Flynt, we split each piece of functionality into smaller, self-contained **feature** bundles.
+
+In most cases, features add global hooks and filters that affect the project on a global level. With this in mind, each feature is built with drag and drop reusability in mind.
```
`drag & drop reusability` reminds me of backend d&d feature. just `reusability`?

emcarru commented:
@domtra: Updated, thanks for the feedback. I've removed `to complete the picture` since that wasn't really adding anything anyway. Though I think it still makes more sense to keep `take ... as an example`, since this is the usage section, and we are trying to explain the concept/usage through an example.
